xdsl/ir.py

```diff
@@ -1102,16 +1103,26 @@ def __hash__(self) -> int:
         return id(self)
 
 
-@dataclass
+@dataclass(init=False)
 class Region(IRNode):
     """A region contains a CFG of blocks. Regions are contained in operations."""
 
-    blocks: list[Block] = field(default_factory=list, init=False)
+    blocks: list[Block]
     """Blocks contained in the region. The first block is the entry block."""
 
-    parent: Operation | None = field(default=None, init=False, repr=False)
+    parent: Operation | None = field(repr=False)
```

Comment:
Is the repr still required here? We are overriding _repr_, right? (Not specific to this PR).

Reply:
Ah, I didn't realise. Maybe it's not required, I didn't pay attention. I'd say best to clean it up in a different PR for sure.